Marlin/src/pins/stm32f1/pins_MKS_ROBIN.h

```diff
@@ -144,71 +187,144 @@
-#define ENABLE_SPI2
+  // Config for Classic UI (emulated DOGM) and Color UI
+  #if ENABLED(TFT_320x240)
+    #define TFT_RESET_PIN                   PF6   // FSMC_RST
+    #define TFT_BACKLIGHT_PIN               PG11
+
+    // MKS Robin TFT with ILI9328
+    //#define TFT_ILI9328
+    #if ENABLED(TFT_ILI9328)
+      #define TFT_DRIVER                 ILI9328
+      #define TFT_BUFFER_SIZE              14400
+      #define ILI9328_COLOR_RGB
+
+      // YV for normal screen mounting
+      //#define ILI9328_ORIENTATION ILI9328_MADCTL_MY | ILI9328_MADCTL_MV
+      // XV for 180° rotated screen mounting
+      #define ILI9328_ORIENTATION ILI9328_MADCTL_MX | ILI9328_MADCTL_MV
+    #endif
+
+    // MKS Robin TFT with R61505
+    //#define TFT_R61505
+    #if ENABLED(TFT_R61505)
+      #define TFT_DRIVER                  R61505
+      #define TFT_BUFFER_SIZE              14400
+      #define R61505_COLOR_RGB
+
+      // YV for normal screen mounting
+      //#define R61505_ORIENTATION R61505_MADCTL_MY | R61505_MADCTL_MV
+      // XV for 180° rotated screen mounting
+      #define R61505_ORIENTATION R61505_MADCTL_MX | R61505_MADCTL_MV
+    #endif
+
+    // MKS Robin TFT with ST7789V
+    //#define TFT_ST7789V
+    #if ENABLED(TFT_ST7789V)
+      #define TFT_DRIVER             TFT_ST7789V
+      #define TFT_BUFFER_SIZE              14400
+      #define ST7789V_COLOR_RGB
+
+      // YV for normal screen mounting
+      //#define ST7789V_ORIENTATION ST7789V_MADCTL_MY | ST7789V_MADCTL_MV
+      // XV for 180° rotated screen mounting
+      #define ST7789V_ORIENTATION ST7789V_MADCTL_MX | ST7789V_MADCTL_MV
+    #endif
+  #endif
+
+#elif HAS_GRAPHICAL_TFT
+
+  #define TFT_RESET_PIN                     PF6
+  #define TFT_BACKLIGHT_PIN                 PG11
 
-#if ENABLED(SDIO_SUPPORT)
-  #define SCK_PIN                           PB13  // SPI2
-  #define MISO_PIN                          PB14  // SPI2
-  #define MOSI_PIN                          PB15  // SPI2
-  #define SD_DETECT_PIN                     PF12  // SD_CD
-#else
-  // SD as custom software SPI (SDIO pins)
-  #define SCK_PIN                           PC12
-  #define MISO_PIN                          PC8
-  #define MOSI_PIN                          PD2
-  #define SS_PIN                            -1
-  #define ONBOARD_SD_CS_PIN                 PC11
-  #define SDSS                              PD2
-  #define SD_DETECT_PIN                     -1
 #endif
 
+//
+// SPI
+//
+#define ENABLE_SPI2
+#define SCK_PIN                             PB13
+#define MISO_PIN                            PB14
+#define MOSI_PIN                            PB15
+// SPI2 is already shared by LCD touch driver and flash
+// SPI2 MOSI (PB15) can be retrieved by soldering job on R17
+// SPI1(PA7) & SPI3(PB5) are not available
+
+//
+// Trinamic TMC2208/2209 UART
+//
 #if HAS_TMC_UART
+
+  // Reduce baud rate for software serial reliability
+  #if HAS_TMC_SW_SERIAL
+    #define TMC_BAUD_RATE                  19200
+  #endif
```

Comment:
I commented about baud rate lower in he file.

I don't think this is going to work, because `HAS_TMC_SW_SERIAL` depends on the pin definitions which come later in the file!